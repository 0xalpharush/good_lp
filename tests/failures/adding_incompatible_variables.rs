use good_lp::variables;

fn main() {
    let mut pb1 = variables!();
    let mut pb2 = variables!();
    let a = pb1.add_variable();
    let b = pb2.add_variable();
    let _result = a + b;
}