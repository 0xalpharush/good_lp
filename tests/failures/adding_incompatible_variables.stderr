error[E0277]: the trait bound `f64: std::convert::From<good_lp::Variable<[closure@$DIR/src/variable.rs:120:60: 120:64]>>` is not satisfied
 --> $DIR/adding_incompatible_variables.rs:8:21
  |
8 |     let _result = a + b;
  |                     ^ the trait `std::convert::From<good_lp::Variable<[closure@$DIR/src/variable.rs:120:60: 120:64]>>` is not implemented for `f64`
  |
  = help: the following implementations were found:
            <f64 as std::convert::From<f32>>
            <f64 as std::convert::From<i16>>
            <f64 as std::convert::From<i32>>
            <f64 as std::convert::From<i8>>
          and 3 others
  = note: required because of the requirements on the impl of `std::convert::Into<f64>` for `good_lp::Variable<[closure@$DIR/src/variable.rs:120:60: 120:64]>`
  = note: required because of the requirements on the impl of `std::convert::From<good_lp::Variable<[closure@$DIR/src/variable.rs:120:60: 120:64]>>` for `good_lp::Expression<[closure@$DIR/src/variable.rs:120:60: 120:64]>`
  = note: required because of the requirements on the impl of `std::convert::Into<good_lp::Expression<[closure@$DIR/src/variable.rs:120:60: 120:64]>>` for `good_lp::Variable<[closure@$DIR/src/variable.rs:120:60: 120:64]>`
  = note: required because of the requirements on the impl of `std::ops::Add<good_lp::Variable<[closure@$DIR/src/variable.rs:120:60: 120:64]>>` for `good_lp::Variable<[closure@$DIR/src/variable.rs:120:60: 120:64]>`
